### ISSUE TITLE
Get Yolo's mAP on Coco

### ISSUE THREAD

**Issue by mmartin56** [pjreddie/darknet#1836]
Hi,

I would like to run Yolov3 on the coco dataset to confirm that I can get the same results as claimed in the yolov3 paper - and that I am running it right. The paper uses coco2014 dataset, which is, according to [the Codalab website](http://cocodataset.org/#guidelines), closed (as well as the 2015 & 2016 datasets).

1) Is there another way to test Yolo on the 2014 dataset than through CodaLab's website?
2) What are Yolo's results on Coco 2017?

Thank you!

**Comment by mmartin56** [comment]
For anyone out there who might having the same question as me, after a couple days search I still couldn't find a way to calculate Yolo's mAP on Coco2014's test set. However Coco2014's validation set is public so it's a great way to verify that you're doing things right (see [that link](https://github.com/AlexeyAB/darknet/issues/2145) for a walk-through). Using AlexeyAB's fork (and weights from yolov3's website), which seems to be better by a few points than pjreddie's branch, I got an mAP of 54.6%. Using yolov3-spp config and weights instead, I obtained 59.3%. That sounds reasonable given yolov3 got 57.9% on Coco2014's test set.

The VOC dataset is also all publicly available, see [there](https://github.com/AlexeyAB/darknet/blob/master/README.md#how-to-calculate-map-on-pascalvoc-2007) for a walk-through and [there](https://github.com/AlexeyAB/darknet/issues/2557) for a good discussion, also using AlexeyAB's fork.

**Comment by alsawaf** [comment]
> 
> 
> For anyone out there who might having the same question as me, after a couple days search I still couldn't find a way to calculate Yolo's mAP on Coco2014's test set. However Coco2014's validation set is public so it's a great way to verify that you're doing things right (see [that link](https://github.com/AlexeyAB/darknet/issues/2145) for a walk-through). Using AlexeyAB's fork (and weights from yolov3's website), which seems to be better by a few points than pjreddie's branch, I got an mAP of 54.6%. Using yolov3-spp config and weights instead, I obtained 59.3%. That sounds reasonable given yolov3 got 57.9% on Coco2014's test set.
> 
> The VOC dataset is also all publicly available, see [there](https://github.com/AlexeyAB/darknet/blob/master/README.md#how-to-calculate-map-on-pascalvoc-2007) for a walk-through and [there](https://github.com/AlexeyAB/darknet/issues/2557) for a good discussion, also using AlexeyAB's fork.

i have downloaded the Val 2017 DataSet with the annotation file , but i still have no idea how use the annotation file , i followed up the tutorial in the link above but it didn't mention anything about how to use the annotation file so do you mind to tell me in details how to get that to work ?

**Comment by eric-tc** [comment]
Hi @mmartin56 

i cannot get the same accuracy on Coco dataset as in the paper. Did you start training from scratch from darknet53.conv.74 and got the same accuracy as in the paper?

**Comment by AlexeyAB** [comment]
You can get much higher accuracy if you train this model: https://raw.githubusercontent.com/AlexeyAB/darknet/master/cfg/csresnext50-panet-spp-original-optimal.cfg
with this pre-trained weights file: https://drive.google.com/open?id=1yb-ZkmzBhO5_wNv-MyQy01jWo_uizLVY
By using this repository: https://github.com/AlexeyAB/darknet#pre-trained-models
